src/random.jl

```diff
@@ -0,0 +1,77 @@
+# random functions that dispatch either to rocRAND or GPUArrays' generic RNG
```

Comment:
Generally my preference is that everything directly under `src/` can always be loaded regardless of binary dependencies, while things under `src/blas/`, `src/rand/`, etc. will be conditionally loaded when the library is available. Could we do that with this file?